Framework/Kernel/src/StringTokenizer.cpp

```diff
@@ -0,0 +1,171 @@
+//
+//  StringTokenizer.cpp
+//  Mantid
+//
+//  Created by Hahn, Steven E. on 1/31/16.
+//
+//
+
+#include "MantidKernel/StringTokenizer.h"
+#include <algorithm>
+#include <functional>
+#include <cassert>
+#include <iterator> //cbegin,cend
+
+namespace {
+
+// implement our own trim function to avoid the locale overhead in boost::trim.
+
+// trim from start
+void ltrim(std::string &s) {
+  s.erase(s.begin(), std::find_if_not(s.begin(), s.end(), ::isspace));
+}
+
+// trim from end
+void rtrim(std::string &s) {
+  s.erase(std::find_if_not(s.rbegin(), s.rend(), ::isspace).base(), s.end());
+}
+
+// trim from both ends
+void trim(std::string &s) {
+  ltrim(s);
+  rtrim(s);
+}
+
+// If the final character is a separator, we need to add an empty string to
+// tokens.
+void emptyFinalToken(const std::string &str, const std::string &delims,
+                     std::vector<std::string> &tokens) {
+  const auto pos = std::find_first_of(str.end() - 1, str.end(), delims.begin(),
```

Comment:
Don't know if it matters performance wise, but since there can't be an empty string here and only the last character of the string is considered, shouldn't it also be safe to do:
```
const auto pos = std::find(delims.cbegin(), delims.cend(), str.back());

if (pos != delims.cend()) {
  tokens.emplace_back()
}
```